app.py

```diff
@@ -244,6 +244,23 @@ def _get_app_path(self, version=None):
         else:
             return self._translate_version_tokens(raw_app_path, version)
 
+    def _get_app_args(self, version=None):
+        """ Return the platform specific app args, performing version substitution. """
+        platform_name = {"linux2": "linux", "darwin": "mac", "win32": "windows"}[sys.platform]
+        raw_app_args = self.get_setting("%s_args" % platform_name, "")
+        if version is None:
+            # there are two reasons version could be None
+            # the first is if versions have not been configured, in which case the raw args are valid
+            # if versions has been configured, then we should expand with the first element in the
+            # list, which will be treated as the default
+            versions = self.get_setting("versions")
+            if versions:
+                return self._translate_version_tokens(raw_app_args, versions[0])
+            else:
+                return raw_app_args
+        else:
+            return self._translate_version_tokens(raw_app_args, version)
+
```

Comment:
Rather than duplicating the code that does the version replacement logic, how about breaking that out to it's own method as well? Something like this (untested):

```python
    def _apply_version_to_setting(self, raw_string, version=None):
        """
        Replace any version tokens  contained in the raw_string with the appropriate version value from 
        the app settings. 

        If version is None, then no version has been specified and we will return the first version 
        from the versions list in the settings. If no versions have been defined in the settings, then 
        we return the raw_string since there's no replacement to do.
        
        :param raw_string: the raw string potentially containing the version tokens (eg. {version}, 
                           {v0}, ...) we will be replacing. This string could represent a number of 
                           things including a path, an args string, etc. 
        :param version: version string to use for the token replacement. 
        :returns: string with version tokens replaced with their appropriate values
        """
        if version is None:
            # there are two reasons version could be None
            # 1. if versions have not been configured, the raw string is assumed valid
            # 2. if versions has been configured, but no specific version was requested, then we 
            #    expand with the first element in the versions list, and use it as the default
            versions = self.get_setting("versions")
            if versions:
               return self._translate_version_tokens(raw_string, versions[0])
            else:
               return raw_string
        else:
            return self._translate_version_tokens(raw_string, version)

    def _get_app_path(self, version=None):
        """ Return the platform specific app path, performing version substitution. """
        platform_name = {"linux2": "linux", "darwin": "mac", "win32": "windows"}[sys.platform]
        raw_app_path = self.get_setting("%s_path" % platform_name, "")
        
        return self._apply_version_to_setting(raw_app_path)

    def _get_app_args(self, version=None):
        """ Return the platform specific app path, performing version substitution. """
        platform_name = {"linux2": "linux", "darwin": "mac", "win32": "windows"}[sys.platform]
        raw_app_args = self.get_setting("%s_args" % platform_name, "")
        
        return self._apply_version_to_setting(raw_app_args)
```